Show appropriate placeholder ‘cover art’ image for events

YatharthROCK opened the pull request:
This PR resolves [CB-186](http://tickets.musicbrainz.org/browse/CB-186). The commit summaries and full messages detail the implementation.

gentlecat commented on critiquebrainz/data/model/review.py:
```diff
@@ -84,6 +85,11 @@ def text(self):
+    @property
+    def is_of_event(self):
+        """Returns whether the review's entity is an event"""
+        return self.entity_type == 'event'
```
I don't think I like this property, way too specific. I'd rather keep things generic here and try not to refer to specific entity type, just compare to `entity_type` where it's used.

gentlecat commented on critiquebrainz/frontend/templates/index.html:
```diff
@@ -35,11 +37,7 @@ <h3 style="margin-bottom:-5px;margin-top:0px;">{{ _('Popular reviews') }}</h3>
-                  .on("error", function () { $(this).attr("src", "/static/img/missing-art.png"); })
-                  .attr("src", "//coverartarchive.org/release-group/{{ review.entity_id }}/front-250");
-            </script>
+            {{ entity_image_setter(review.entity_id, review.is_of_event) }}
```
Can you make this macro more independent by making it add `<img />` tag as well? And a better name then would be just `cover_art`.

gentlecat commented on critiquebrainz/frontend/templates/macros.html:
```diff
@@ -35,3 +35,11 @@
   {% endif %}
 {% endmacro %}
+
+{% macro entity_image_setter(id, is_event) %}
```
So here I'd rather have `entity_type` argument and do comparison inside the macro, as needed.

gentlecat commented on critiquebrainz/frontend/templates/macros.html:
```diff
@@ -35,3 +35,11 @@
+  <script>
+    $("img").last().on("error", function () {
+      $(this).attr("src", "{{ '/static/img/event-placeholder.png' if is_event else '/static/img/missing-art.png' }}");
+    }).attr("src", "//coverartarchive.org/release-group/{{ id }}/front-250");
```
There's no point in trying to get release group cover art for events.

gentlecat commented:
Also, it might be better to switch to SVG. Did you draw it yourself?

YatharthROCK commented:
@gentlecat Thanks for the feedback! I've incorporated it.

Regarding the image: I got it from [here](https://pixabay.com/en/location-destination-shadow-pin-948510/). It's licensed under CC0 Public Domain, which is compatible with GPLv2. It didn't require attribution, and I had not thought I should add an entry to `LICENSE.md` or add a LICENSE file in the `static/imgs` directory. Should I have?

gentlecat commented:
Is there an SVG version? It would be nice to adjust colors and maybe remove the shadow to make it look more subtle.

YatharthROCK commented:
@gentlecat There is not an SVG version; I had recolored the image as a PNG by adjusting saturation and brightness/exposure. However, I found [an image](https://pixabay.com/en/location-tracker-map-position-1093169/) that did have an SVG version. I added and removed elements and precisely matched the color values of the background. Here is a preview of the event and RG cover art together:
![](http://i.imgur.com/U4NsKDd.png)

Does it look fine? Also, should the SVG be served directly or a PNG export of it?

YatharthROCK commented:
I feel the missing cover art image needs to be made less subtle as it is nearly indistinguishable from a grey circle. I didn't see an SVG version in the repo; perhaps a ticket could be made about this?

gentlecat commented:
I think it's just right. It shouldn't stand out.

Regarding your place icon, can you remove circle in the background and just leave place marker (or whatever it's called) and make it the same color as current missing cover art image?

gentlecat commented on critiquebrainz/frontend/templates/macros.html:
```diff
@@ -35,3 +35,19 @@
   {% endif %}
 {% endmacro %}
+
+{% macro coverart(entity_id, entity_type, attributes='') %}
```
`cover_art`

YatharthROCK replied:
Resolved in 7e8843f

gentlecat commented on critiquebrainz/frontend/templates/macros.html:
```diff
@@ -35,3 +35,19 @@
+{% macro coverart(entity_id, entity_type, attributes='') %}
+  {# Note: when passing in `attributes`, don't quote attribute values; they will be added automatically #}
+  {% if entity_type == 'event' %}
+    <img src="/static/img/event-placeholder.png" {{ attributes }} />
```
Why not SVG? You added it into the repo.

YatharthROCK replied:
@gentlecat I must have not realized this after I switched to using a SVG from a PNG; thanks! Resolved in cea06b1

YatharthROCK commented:
@gentlecat Resolved in d71a394

gentlecat commented on critiquebrainz/frontend/templates/review/modify/release_group.html:
```diff
@@ -17,10 +19,5 @@
-        .on("error", function () { $(this).attr("src", "/static/img/missing-art.png"); })
-        .attr("src", "//coverartarchive.org/release-group/{{ entity['id'] }}/front-250");
-  </script>
+  {{acover_art(entity['id'], 'release_group', 'id=cover-art class=img-responsive style=max-height:120px;max-width:120px;') }}
```
Looks like a typo there.

YatharthROCK replied:
@gentlecat Oops, sorry that slipped through; I'll check the final diff more thoroughly next time. Resolved in 65f3275

gentlecat commented:
Nice. I think that looks good now.
